numpyro/contrib/funsor/discrete.py

```diff
@@ -71,19 +71,7 @@ def _sample_posterior(
     for name, node in sample_tr.items():
         if node["type"] != "sample":
             continue
-        if node["is_observed"]:
-            # "observed" values may be collapsed samples that depend on enumerated
-            # values, so we have to slice them down
-            # TODO this should really be handled entirely under the hood by adjoint
-            output = funsor.Reals[node["fn"].event_shape]
-            value = funsor.to_funsor(
-                node["value"], output, dim_to_name=node["infer"]["dim_to_name"]
-            )
-            value = value(**sample_subs)
-            node["value"] = funsor.to_data(
-                value, name_to_dim=node["infer"]["name_to_dim"]
-            )
-        else:
+        if node["infer"].get("enumerate") == "parallel":
```

Comment:
There are non-enumerated latent variables in the GMM tutorial that trigger the `else` branch here but since these variables are not enumerated `_get_support_value(log_measure, name)` will give an error.

I think we only need to get support values for enumerated variables. For other variables adjoint should just return their sampled/observed values. I don't know what has changed in funsor since this code was written but it seems like adjoint handles well those cases under the hood (see `# TODO this should really be handled entirely under the hood by adjoint` comment).